src/audio/pipeline.c

```diff
@@ -223,6 +222,22 @@ static void pipeline_trigger_sched_comp(struct pipeline *p,
 			}
 		}
 		break;
+	case COMP_TRIGGER_RELEASE:
```

Comment:
Ok, this is in the right direction, but instead of scheduling the pipeline to fix this issue with old data we should actually reset the pipeline buffers (i.e. clear the buffers to 0s and reset their R/W ptr to initial start positions). This way we recover without XRUN and we do not send stale/old capture data to the host (as pause can be longer than a few seconds).

Reply:
yes, should be merged. Capture stop/start and pause/release have same flow, i.e we discard old data and reset R/W pts.

Reply:
ok, I will send the next version.

Reply:
no, playback should not clear the buffers on pause, only on stop

Reply:
I think it is not big issue that we don't clear buffers on capture pause? in that way, those captured data(in buffer) will be written to application somewhat late only, those data still make sense.

Reply:
no, that's wrong. if you are recording a conversation and then pause for 5 seconds you do not expect the first few periods after resume to contain audio from 5 secs ago as they will be out of context and may even cause audible artifacts.

Reply:
but think from another side, if we reset buffers there may be milliseconds data lost, it is also another kind of artifacts?
and, the case you listed above may happen for playback also, considering live playback, if we don't reset buffer for pause on playback, we may also hear out of context audio which may also cause audible artifacts.

Reply:
What about another solution:
Still scheduling the pipeline to fix this bug, but when pause the pipeline, we clear the buffer to 0s. do not reset the R/W ptr.